src/cmap/establish/handshake/mod.rs

```diff
@@ -0,0 +1,192 @@
+#[cfg(test)]
+mod test;
+
+use bson::{doc, Bson, Document};
+use lazy_static::lazy_static;
+use os_info::{Type, Version};
+
+use crate::{
+    cmap::{options::ConnectionPoolOptions, Command, Connection, StreamDescription},
+    error::Result,
+    is_master::IsMasterReply,
+};
+
+#[derive(Clone, Debug)]
```

Comment:
Rather than having to deal with setting nested values in a document, I figured it was easier to just use a struct for all of the client metadata and then convert to a document only once we've computed all the values.